### DISCUSSION TITLE
Can't use x-transition

### DISCUSSION THREAD

**Discussion by rafli-sss** [alpinejs/alpine#4021] [1. Help] [answered]
I have another component that also uses `x-show` and `x-transition` it works. However, I got `Uncaught TypeError: directiveStorageMap[stage] is not a function` in console with this component. If I remove the `x-transition` the error disappears

```blade
<form
 class="space-y-3"
 action=""
 method="GET"
 x-data="{
   gradeInputChanged: false,
   majorInputChanged: false,
  }"
>
{{-- other input --}}
<x-input-group
  class="max-sm:col-span-2"
  input-name="index"
  input-id="index-input"
  x-show="gradeInputChanged && majorInputChanged"
  x-transition
>
  <x-label>Index</x-label>
  <x-select>
    <option value="">All</option>
    @foreach ($indexes as $index)
      <option @selected(request()->query('index') == $index)>
        {{ $index }}
      </option>
    @endforeach
  </x-select>
</x-input-group>
</form>
```

```html
<!-- x-input-group is rendered like this in devtools -->
<div
  class="max-sm:col-span-2"
  x-show="gradeInputChanged && majorInputChanged"
  x-transition="x-transition"
>
  <!-- label and select input here -->
</div>
```

**Comment by shubdragon** [comment]
should we do  `x-transition`  this ✅  `x-transition="x-transition"` not this ❌ Isn't it ? after changing this thing your problem will be solved

[Source](https://alpinejs.dev/directives/transition)

> Tip add `x-cloak` with it, if you are seeing flickering effect on page load

**Reply by rafli-sss** [reply]
Of course, I only write `x-transition`. `x-transition="x-transition"` is what the browser renders (as seen through Inspect Element devtools).

**Reply by shubdragon** [reply, marked as answer]
Okay, you are using `x-transition` on a x-component this is the issue, is you directly apply inside the component it will not be a issue than, or you could say it is as a bug.

But there is a workaround for this you could use on your included component `x-bind="someTransition"` and on `x-data="{someTransition: ['x-transition']}` inside your component . This way it will correctly render on browser, its just a idea you could do for temporary working.

Similiar to this https://alpinejs.dev/directives/bind#bind-directives

**Reply by rafli-sss** [reply]
Thank you. I think you are right. The `x-transition` attribute on the blade component will be rendered as `x-transition="x-transition"` (wrong) instead of `x-transition=""` (correct). Now, I use `x-transition` with CSS classes.
